include/oneapi/tbb/concurrent_queue.h

```diff
@@ -135,37 +136,42 @@ class concurrent_queue {
     }
   
     concurrent_queue& operator=( const concurrent_queue& other ){
-      if (my_queue_representation != other.my_queue_representation) {
-	my_queue_representation = other.my_queue_representation;
-      }
-      return *this;
+        //TODO: implement support for std::allocator_traits::propogate_on_container_copy_assignment
+        if (my_queue_representation != other.my_queue_representation) {
+            clear();
+	    my_queue_representation->assign(*other.my_queue_representation, my_allocator, copy_construct_item);
+	}
+	return *this;
     }
 
     concurrent_queue& operator=( concurrent_queue&& other ){
-      if (my_queue_representation != other.my_queue_representation) {
-	my_queue_representation = std::move(other.my_queue_representation);
-      }
-      return *this;
+        //TODO: implement support for std::allocator_traits::propogate_on_container_move_assignment
+        if (my_queue_representation != other.my_queue_representation) {
+            my_queue_representation = std::move(other.my_queue_representation);
```

Comment:
Since `my_queue_representation` is a pointer, such a move is incorrect, because it will just copy pointers and result in two queues with the same representation. 
The correct move assignment would be something like:
```
if (my_allocator == other.my_allocator) {
    my_queue_representation = other.my_queue_representation;
    other.my_queue_representation = nullptr;
} else {
    // Deep per-element move, e.g. with assign (again if assign re-allocates everything)
    clear();
    my_queue_representation->assign(*other.my_queue_representation, my_allocator, move_construct_item);
}
```

For move construction/assignment, we should reuse existing move support tests that covers tricky use-cases with equal/unequal allocators. See how it is done for other containers as a reference.